libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestRequest.m

```diff
@@ -164,6 +165,11 @@ - (NSURLRequest *)prepareRequestForSend {
             [self.request setValue:bearer forHTTPHeaderField:@"Authorization"];
         }
 
+        // Sets Mobile SDK user agent on REST API requests, if it hasn't been set already by .
+        if (self.request.allHTTPHeaderFields && self.request.allHTTPHeaderFields.allKeys && ![self.request.allHTTPHeaderFields.allKeys containsObject:@"User-Agent"]) {
```

Comment:
Not to necessarily change the code, but the following test:
```obj-c
if (![self.request.allHTTPHeaderFields.allKeys containsObject:@"User-Agent"]) {
}
```
is all that's necessary in Objective-C.  Any of the upstream properties being `nil` will automatically evaluate to 'false' from that evaluation.